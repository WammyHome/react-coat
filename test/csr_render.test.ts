import {mount} from "enzyme";
import client from "./client";

test("spa success:/videos?search={title:success}", done => {
  expect.assertions(2);
  const actions: string[] = [];
  const logerMiddleware = ({dispatch}: {dispatch: Function}) => (next: Function) => (originalAction: {type: string}) => {
    actions.push(originalAction.type);
    return next(originalAction);
  };
  const store = client('/videos?search={"title":"success"}', [logerMiddleware], app => {
    const wrapper = mount(app);
    setTimeout(() => {
      expect({html: wrapper.html(), data: store.getState()}).toMatchSnapshot();
      expect(actions).toEqual(["app/INIT", "app/LOADING", "app/UPDATE", "videos/INIT", "app/LOADING", "@@framework/VIEW_INVALID", "videos/searchList", "videos/LOADING", "videos/UPDATE", "videos/LOADING"]);
      done();
    }, 1000);
  });
});
