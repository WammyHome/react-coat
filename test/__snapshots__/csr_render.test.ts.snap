// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`spa success:/videos?search={title:success} 1`] = `
Object {
  "data": Object {
    "app": Object {
      "curUser": Object {
        "avatarUrl": "/imgs/1.jpg",
        "hasLogin": true,
        "uid": "1",
        "username": "test",
      },
      "isModule": true,
      "loading": Object {
        "global": "Stop",
        "login": "Stop",
      },
    },
    "router": Object {
      "action": "POP",
      "location": Object {
        "hash": "",
        "pathname": "/videos",
        "search": "?search={%22title%22:%22success%22}",
        "state": undefined,
      },
    },
    "videos": Object {
      "isModule": true,
      "listItems": Array [
        Object {
          "id": "1",
          "title": "item1",
        },
        Object {
          "id": "2",
          "title": "item2",
        },
      ],
      "listSearch": Object {
        "page": 1,
        "pageSize": 10,
        "title": "success",
      },
      "listSummary": Object {
        "page": 1,
        "pageSize": 10,
        "totalItems": 10,
        "totalPages": 1,
      },
      "loading": Object {
        "global": "Stop",
      },
    },
    "views": Object {
      "app": Object {
        "Main": 1,
      },
      "videos": Object {
        "Main": 1,
      },
    },
  },
  "html": "<div><div class=\\"videos\\"><ul><li>video-item1</li><li>video-item2</li></ul><div class=\\"pagination\\">共10条，第1/1页</div></div></div>",
}
`;
